### PR TITLE
Elite progress

### PR DESCRIPTION
**Pull request by bettse**
# What's new

- The idea is a progress bar for picopass elite dictionary

# Verification 

- Test a card and see it shows progress bar

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Review by skotopes — APPROVED** [review]

**Comment by skotopes** [comment]
Elite dict attack is quite difficult to exit, it immediately fails if card is not present.

File: applications/external/picopass/picopass_worker.c [hunk 1]
```diff
@@ -460,6 +458,129 @@ ReturnCode picopass_write_block(PicopassBlock* AA1, uint8_t blockNo, uint8_t* ne
     return ERR_NONE;
 }
 
+void picopass_worker_elite_dict_attack(PicopassWorker* picopass_worker) {
```
Comment (skotopes) [inline applications/external/picopass/picopass_worker.c:461]: I think it should ensure that card presence before starting main routine.

**Review by skotopes — CHANGES_REQUESTED** [review]

File: applications/external/picopass/picopass_worker.c [hunk 2]
```diff
@@ -460,6 +458,136 @@ ReturnCode picopass_write_block(PicopassBlock* AA1, uint8_t blockNo, uint8_t* ne
...
+        uint8_t* div_key = AA1[PICOPASS_KD_BLOCK_INDEX].data;
+
+        loclass_iclass_calc_div_key(csn, key, div_key, elite);
+        loclass_opt_doReaderMAC(ccnr, div_key, mac);
+
+        err = rfalPicoPassPollerCheck(mac, &chkRes);
+        if(err == ERR_NONE) {
+            FURI_LOG_I(TAG, "Found key");
+            memcpy(pacs->key, key, PICOPASS_BLOCK_LEN);
+            if(nextState != PicopassWorkerEventFail) {
```
Comment (skotopes) [inline applications/external/picopass/picopass_worker.c]: Expression 'nextState != PicopassWorkerEventFail' is always true.

**Review by skotopes — CHANGES_REQUESTED** [review]

File: applications/external/picopass/picopass_worker.c (hunk 2 above)
Comment (skotopes) [reply applications/external/picopass/picopass_worker.c]: Overall it looks like this routine delivering PicopassWorkerEventSuccess even if it failed.

Comment (bettse) [reply applications/external/picopass/picopass_worker.c]: Yeah, I'd meant to remove all of nextState and forgotten/missed some spots

**Review by skotopes — APPROVED** [review]

**Pull request merged**